### PR TITLE
Fix Enum Configuration variables

### LINKED ISSUE #134: Can't use configuration variable enum attribute
Looks like the enum attribute is broken
the schema define `enum` 
https://github.com/env0/terraform-provider-env0/blob/76b088240cc2db75ab2b9eda6cb40b26ce2c7446/env0/resource_configuration_variable.go#L66-L73

but in the create code we look for `enum_values`
https://github.com/env0/terraform-provider-env0/blob/76b088240cc2db75ab2b9eda6cb40b26ce2c7446/env0/resource_configuration_variable.go#L119-L122

Setting them both to `enum` or `enum_values` results with
```
panic: interface conversion: interface {} is []interface {}, not []string
```

### PR DESCRIPTION
**Pull request by eranelbaz**
### Issue & Steps to Reproduce / Feature Request
resolves #134

### Solution
`resource`
when you are reading using `d.GetOk` it return value which type is undefined - `interface{} | []interface{}` so we can't just cast it to `[]string`
we need to cast to slice of interface and then iterate over and cast each element

`data`
just get and set to `enum` from the `schema`

added UT, example and update the docs

### CODE REVIEW

File: examples/resources/env0_configuration_variable/resource.tf [hunk 1]
```diff
@@ -2,3 +2,13 @@ resource "env0_configuration_variable" "example" {
   name  = "ENVIRONMENT_VARIABLE_NAME"
   value = "example value"
 }
+
+resource "env0_configuration_variable" "drop_down" {
+  name  = "ENVIRONMENT_VARIABLE_NAME"
```
Comment (avnerenv0) [inline examples/resources/env0_configuration_variable/resource.tf:7]: this won't really work because this file already has `ENVIRONMENT_VARIABLE_NAME`, I know that's a small semantic but let's try to keep the examples working.

File: env0/data_configuration_variable.go [hunk 2]
```diff
@@ -77,6 +77,15 @@ func dataConfigurationVariable() *schema.Resource {
 				Description: "scope of the variable",
 				Computed:    true,
 			},
+			"enum": {
+				Type:        schema.TypeList,
+				Description: "limit possible values to values from this list",
```
Comment (avnerenv0) [inline env0/data_configuration_variable.go:82]: Since it's the `data`, it's just for reading

```suggestion
				Description: "possible values of this variable",
```

File: env0/resource_configuration_variable.go [hunk 3]
```diff
@@ -130,6 +131,25 @@ func resourceConfigurationVariableCreate(ctx context.Context, d *schema.Resource
...
 
+func getEnum(d *schema.ResourceData, value string) ([]string, diag.Diagnostics) {
+	var enumValues []interface{} = nil
+	var actualEnumValues []string = nil
+	if specified, ok := d.GetOk("enum"); ok {
+		enumValues = specified.([]interface{})
+		var valueExists = false
+		for _, val := range enumValues {
+			actualEnumValues = append(actualEnumValues, val.(string))
+			if val == value {
```
Comment (avnerenv0) [inline env0/resource_configuration_variable.go:142]: the use of `val` AND `value` in the same spot is a little confusing 
can you make that `if enumValue == selectedValue` or something like that?

File: tests/integration/003_configuration_variable/main.tf [hunk 4]
```diff
@@ -32,10 +32,3 @@ data "env0_configuration_variable" "tested1" {
   depends_on = [env0_configuration_variable.tested1]
 }
 
-output "tested1_value" {
```
Comment (avnerenv0) [inline tests/integration/003_configuration_variable/main.tf:35]: why did you remove this ?
If anything I'd like to see you add an enum variable here.

**Review by avnerenv0 — COMMENTED** [review]
Looks good, just concerned about removing integration tests

File: examples/resources/env0_configuration_variable/resource.tf (hunk 1 above)
Comment (eranelbaz) [reply examples/resources/env0_configuration_variable/resource.tf]: ac80073f

File: env0/data_configuration_variable.go (hunk 2 above)
Comment (eranelbaz) [reply env0/data_configuration_variable.go]: ac80073f

File: env0/resource_configuration_variable.go (hunk 3 above)
Comment (eranelbaz) [reply env0/resource_configuration_variable.go]: 42577a85

File: tests/integration/003_configuration_variable/main.tf (hunk 4 above)
Comment (eranelbaz) [reply tests/integration/003_configuration_variable/main.tf]: 2290182b
added enum check

**Review by yaronya — COMMENTED** [review]
👍 
Left some comments

File: docs/resources/configuration_variable.md [hunk 5]
```diff
@@ -17,6 +17,15 @@ resource "env0_configuration_variable" "example" {
   name  = "ENVIRONMENT_VARIABLE_NAME"
   value = "example value"
 }
+
+resource "env0_configuration_variable" "drop_down" {
+  name  = "ENVIRONMENT_VARIABLE_DROP_DOWN"
+  value = "first option"
```
[the remark this answers is not in the document]
Comment (eranelbaz) [reply docs/resources/configuration_variable.md:23]: we throw exception

File: env0/data_configuration_variable.go [hunk 6]
```diff
@@ -77,6 +77,15 @@ func dataConfigurationVariable() *schema.Resource {
 				Description: "scope of the variable",
 				Computed:    true,
 			},
+			"enum": {
+				Type:        schema.TypeList,
+				Description: "possible values of this variable",
+				Computed:    true,
```
[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/data_configuration_variable.go:83]: since we get that from BE

File: env0/resource_configuration_variable.go [hunk 7]
```diff
@@ -130,6 +131,25 @@ func resourceConfigurationVariableCreate(ctx context.Context, d *schema.Resource
...
+		var valueExists = false
+		for _, enumValue := range enumValues {
+			actualEnumValues = append(actualEnumValues, enumValue.(string))
+			if enumValue == selectedValue {
+				valueExists = true
+			}
+		}
+		if !valueExists {
+			return nil, diag.Errorf("value - '%s' is not one of the enum options %v", selectedValue, actualEnumValues)
+		}
```
[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/resource_configuration_variable.go:148]: idk if you can do that we have `AtLeastOneOf` but it part of static set of values

File: env0/resource_configuration_variable.go [hunk 8]
```diff
@@ -130,6 +131,25 @@ func resourceConfigurationVariableCreate(ctx context.Context, d *schema.Resource
 	return nil
 }
 
+func getEnum(d *schema.ResourceData, selectedValue string) ([]string, diag.Diagnostics) {
+	var enumValues []interface{} = nil
+	var actualEnumValues []string = nil
```
[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/resource_configuration_variable.go:136]: 3eb1adf0

File: env0/resource_configuration_variable.go [hunk 9]
```diff
@@ -130,6 +131,25 @@ func resourceConfigurationVariableCreate(ctx context.Context, d *schema.Resource
...
+	if specified, ok := d.GetOk("enum"); ok {
+		enumValues = specified.([]interface{})
+		var valueExists = false
```
[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/resource_configuration_variable.go]: d129d5f8

File: env0/resource_configuration_variable.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (eranelbaz) [reply env0/resource_configuration_variable.go:148]: Couldn't find use of `ValidateDiagFunc` with another attribute values

**Review by yaronya — APPROVED** [review]

**Comment by avnerenv0** [comment]
[Can't use enum attribute with configuration variable](https://trello.com/c/zGl1kkUF/4028-cant-use-enum-attribute-with-configuration-variable)
